File: src/osx/prtscn_osx.cpp
#include <CoreGraphics/CoreGraphics.h>
#include "prtscn_osx.h"

CGImageRef getScreen(const int x, const int y, int width, int height)
{
	if (width == 0)
	{
		width = CGDisplayPixelsWide(CGMainDisplayID());
	}
	if (height == 0)
	{
		height = CGDisplayPixelsHigh(CGMainDisplayID());
	}
	CGRect rect = CGRectMake(x, y, width, height);
	CGImageRef image_ref = CGDisplayCreateImageForRect(CGMainDisplayID(), rect);
	CGImageRelease(image_ref);
	return image_ref;
}

File: src/osx.1/prtscn_osx.m
#include <Foundation/Foundation.h>
#include <CoreServices/CoreServices.h>
#include "prtscn_osx.h"

// CGImageRef转换为NSData
NSData * CGimageRefToNSData(CGImageRef imageRef)
{
	CFMutableDataRef dataRef = CFDataCreateMutable(NULL, 0);
	CGImageDestinationRef destinationRef = CGImageDestinationCreateWithData(dataRef, kUTTypePNG, 1, NULL);
	CGImageDestinationAddImage(destinationRef, imageRef, nil);
	if(!CGImageDestinationFinalize(destinationRef)) {
		NSLog(@"Failed to write Image");
	}
	NSData *data = (NSData *)dataRef;
	CFRelease(dataRef);
	return data;
}

CFDataRef getScreen(const int x, const int y, const int width, const int height)
{	
	CGRect rect = CGRectMake(x, y, width, height);
	CGImageRef imageRef = CGDisplayCreateImageForRect(CGMainDisplayID(), rect);
	CFDataRef rawData = CGDataProviderCopyData(CGImageGetDataProvider(imageRef));
	// NSData * data = CGimageRefToNSData(imageRef);
	// unsigned int length = [data length];
	// NSString *base64Encoded = [data base64EncodedStringWithOptions:0];
	// const char *base64 = [base64Encoded UTF8String];
	// CGImageRelease(imageRef);

	return rawData;
}
